### ISSUE TITLE
hard crash with average traces

### ISSUE THREAD

**Issue by ggggggggg** [usnistgov/microscope#27]
Steps to reproduce:
1. turn off all triggering in dastard commander
2. start new microscope
3. click "average traces" checkbox
4. add "Ch 1" to plots by clicking the up arrow in the spin corresponding to the black trace (Curve A)

Kelsey thinks it might be a divide by zero error, which seems pretty plausible.

**Comment by joefowler** [comment]
Weirdly, I could reproduce this one, and then never again. Even a return to the original code, a `make clean; make` and a restart did not cause the crash to return. Thoughts?

The code in `pulseRecord *pulseHistory::meanRecord(int nAverage)` appears to avoid all possible divide-by-zero. I think in fact the error was some kind of null pointer dereference, or closer to that. At least, _my_ error was a SIGSEGV.

**Comment by ggggggggg** [comment]
I don't have any insight into why you could reproduce once but not again, but when I was testing it with Kelsey we were able to repeat it 5-10x in a row.

**Comment by joefowler** [comment]
I think I repeated a few times in a row. It was only after I changed some things and recompiled that the crash stopped happening, even after a stashed my changes and rebuilt the previous version. Hmm.

My plan: remove a bunch of compiler warnings that were not previously seen (why? maybe because I'm using gcc version 9 now?), push to master, and try to test on another machine whether the problem is or isn't fixed.
